software/follow_ball_depth.py

```diff
@@ -0,0 +1,381 @@
+import image_processor
+import camera
+import motion
+import cv2
+import time
+import threading
+import referee_command
+import json
+import pandas as pd
+import numpy as np
+import math
+from realsense_depth import *
+
+exitFlag = 0
+global speedz
+global speedx
+global state
+state = 'find_ball'
+
+speed_df = pd.read_csv('speed_magenta.csv')
+
+
+
+def speed_thrower(dist):
+    
+    #speed = min(int(0.316*dist + 450),1900)
+
+    speed = min(int(0.330*dist + 415),1900)
+
+    """
+    if dist <= 1500:
+        speed = int(0.316*dist + 450)
+   
+    elif dist > 1500 and dist < 2300:
+        speed = int(0.316*dist + 450)
+
+    elif dist >= 2300:
+       speed = int(0.316*dist + 450)
+    """
+    return speed
+
+def speed_limitation(speed,direction):
```

Comment:
If you need hard limiters write it in the driving code that us responsive for omni motion